### PR TITLE
Graph examples

### PR DESCRIPTION
**Pull request by FlorentinD**
## What changes were proposed in this pull request?

Added examples using the spark-graph module

### CODE REVIEW

**Comment by FlorentinD** [comment]
Currently the examples have problems importing `spark.cypher` as well as `spark.graph`.
This occured only after rebasing this branch onto the latest progress of the poc branch.
Probably a problem with the POM (again). 
@knutwalker maybe you could help me out? :slightly_smiling_face:

File: graph/api/src/main/scala/org/apache/spark/graph/api/RelationshipFrameBuilder.scala [hunk 1]
```diff
@@ -27,7 +27,7 @@ import scala.collection.JavaConverters._
  * @param df DataFrame containing a single relationship in each row
  * @since 3.0.0
  */
-final class RelationshipFrameBuilder(val df: DataFrame) {
+final class RelationshipFrameBuilder(val df: Dataset[Row] {
```
Comment (knutwalker) [inline graph/api/src/main/scala/org/apache/spark/graph/api/RelationshipFrameBuilder.scala]:
```suggestion
final class RelationshipFrameBuilder(val df: Dataset[Row]) {
```

File: examples/pom.xml [hunk 2]
```diff
@@ -102,17 +120,145 @@
...
-      <scope>provided</scope>
+      <scope>compile</scope>
     </dependency>
     <dependency>
       <groupId>com.github.scopt</groupId>
       <artifactId>scopt_${scala.binary.version}</artifactId>
       <version>3.7.1</version>
+      <groupId>log4j</groupId>
+      <artifactId>log4j</artifactId>
+      <scope>compile</scope>
```
Comment (knutwalker) [inline examples/pom.xml:134]: There's probably a conflict resolution gone wrong. The dependency of log4j needs their onw dependency/ block

Comment (knutwalker) [inline examples/pom.xml:134]:
```suggestion
    </dependency>
    <dependency>
      <groupId>log4j</groupId>
      <artifactId>log4j</artifactId>
      <scope>compile</scope>
```

Comment (knutwalker) [reply examples/pom.xml:134]: suggestion might not work though, github seems to only want to delete one line

**Review by knutwalker — COMMENTED** [review]
@FlorentinD got it to work in https://github.com/FlorentinD/spark/pull/2

File: examples/pom.xml (hunk 2 above)
Comment (FlorentinD) [reply examples/pom.xml:134]: ah right, didn't see that

**Pull request merged**